plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftUnload.java

```diff
@@ -0,0 +1,239 @@
+                //  lineitem and orders need to be re-created.
+                .setInitialTables(REQUIRED_TPCH_TABLES)
+                .build();
+    }
+
+    private RedshiftSplitManager createSplitManager()
+    {
+        DriverConnectionFactory driverConnectionFactory = DriverConnectionFactory.builder(new Driver(), JDBC_URL, new ExtraCredentialProvider(Optional.empty(), Optional.empty(), new StaticCredentialProvider(Optional.of(JDBC_USER), Optional.of(JDBC_PASSWORD))))
+                .build();
+        RedshiftClient redshiftClient = new RedshiftClient(
+                new BaseJdbcConfig().setConnectionUrl(JDBC_URL),
+                redshiftUnloadConfig,
+                driverConnectionFactory,
+                new JdbcStatisticsConfig(),
+                new DefaultQueryBuilder(NONE),
+                new DefaultIdentifierMapping(),
+                NONE);
+
+        return new RedshiftSplitManager(
+                redshiftClient,
+                new DefaultQueryBuilder(NONE),
+                NONE,
+                new JdbcSplitManager(redshiftClient),
+                redshiftUnloadConfig,
+                new S3FileSystemFactory(OpenTelemetry.noop(), new S3FileSystemConfig()
+                        .setAwsAccessKey(AWS_ACCESS_KEY)
+                        .setAwsSecretKey(AWS_SECRET_KEY)
+                        .setRegion(AWS_REGION)
+                        .setStreamingPartSize(DataSize.valueOf("5.5MB")), new S3FileSystemStats()),
+                Executors.newFixedThreadPool(1));
+    }
+
+    @Test
+    void testUnloadFlow()
+            throws Exception
+    {
+        JdbcTableHandle table =
+                new JdbcTableHandle(
+                        new SchemaTableName(TEST_SCHEMA, "nation"),
+                        new RemoteTableName(Optional.of(TEST_DATABASE), Optional.of(TEST_SCHEMA), "nation"),
+                        Optional.empty());
+
+        ConnectorSession session = createUnloadSession();
+        ConnectorSplitSource splitSource = redshiftSplitManager.getSplits(null, session, table, DynamicFilter.EMPTY, Constraint.alwaysTrue());
+        assertThat(splitSource).isInstanceOf(RedshiftUnloadSplitSource.class);
+
+        ConnectorSplitSource.ConnectorSplitBatch connectorSplitBatch = splitSource.getNextBatch(2).get();
+        assertThat(connectorSplitBatch.isNoMoreSplits()).isTrue();
+        assertThat(connectorSplitBatch.getSplits()).hasSize(2);
+        assertThat(connectorSplitBatch.getSplits()).allMatch(split -> split instanceof RedshiftUnloadSplit);
+        List<RedshiftUnloadSplit> redshiftUnloadSplits = connectorSplitBatch.getSplits().stream().map(split -> ((RedshiftUnloadSplit) split)).collect(toImmutableList());
+        verifyS3UnloadedFiles(redshiftUnloadSplits);
+        removeS3UnloadedFiles(redshiftUnloadSplits);
+    }
+
+    @Test
+    void testUnloadFlowFallbackToJdbc()
+            throws Exception
+    {
+        // Fallback to JDBC as limit clause is not supported by UNLOAD
```

Comment:
Why does this require a special test here ?
Shouldn't verifying that LIMIT queries work in normal query runner tests with unload enabled be enough ?

Reply:
ya that would be e2e test but I wanted to add an explicit assert to verify that, for unsupported scenario's JDBC flow is called. I can add a limit query execution through queryrunner at the end of the test body. Is that fair or we just need to test limit query through query runner?